fv3core/stencils/fillz.py

```diff
@@ -61,101 +69,70 @@ def fix_interior(
 
 
 def compute(dp2, tracers, im, km, nq, jslice):
     # Same as above, but with multiple tracer fields
-    shape = tracers[utils.tracer_variables[0]].shape
-    i1 = grid().is_
+    i1 = spec.grid.is_
     js = jslice.start
-    j_extent = jslice.stop - jslice.start
-    orig = (i1, js, 0)
-    zfix = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
-    upper_fix = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
-    lower_fix = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
+    jext = jslice.stop - jslice.start
+
+    tracer_list = [tracers[q] for q in utils.tracer_variables[0:nq]]
+    shape = tracer_list[0].shape
+    shape_ij = shape[0:2]
+
     dm = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
     dm_pos = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
-    fac = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
+    upper_fix = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
+    lower_fix = utils.make_storage_from_shape(shape, origin=(0, 0, 0))
+    zfix = utils.make_storage_from_shape(shape_ij, dtype=np.int, origin=(0, 0))
+    fac = utils.make_storage_from_shape(shape_ij, origin=(0, 0))
+    sum0 = utils.make_storage_from_shape(shape_ij, origin=(0, 0))
+    sum1 = utils.make_storage_from_shape(shape_ij, origin=(0, 0))
     # TODO: Implement dev_gfs_physics ifdef when we implement compiler defs.
 
-    for q in utils.tracer_variables[0:nq]:
-        # reset fields
-        zfix[:] = 0.0
-        fac[:] = 0.0
-        lower_fix[:] = 0.0
-        upper_fix[:] = 0.0
-        fix_top(tracers[q], dp2, dm, origin=orig, domain=(im, j_extent, 2))
-        fix_interior(
-            tracers[q],
+    for tracer in tracer_list:
+        fix_tracer(
+            tracer,
             dp2,
-            zfix,
-            upper_fix,
-            lower_fix,
             dm,
             dm_pos,
-            origin=(i1, js, 0),
-            domain=(im, j_extent, km),
-        )
-        fix_bottom(
-            tracers[q],
-            dp2,
             zfix,
             upper_fix,
             lower_fix,
-            dm,
-            dm_pos,
-            origin=(i1, js, km - 2),
-            domain=(im, j_extent, 2),
-        )
-        fix_cols = utils.sum(zfix[:], axis=2)
-        zfix[:] = utils.repeat(fix_cols[:, :, np.newaxis], km + 1, axis=2)
-        sum0 = utils.sum(dm[:, :, 1:], axis=2)
-        sum1 = utils.sum(dm_pos[:, :, 1:], axis=2)
-        adj_factor = utils.zeros(sum0.shape)
-        adj_factor[sum0 > 0] = sum0[sum0 > 0] / sum1[sum0 > 0]
-        fac[:] = utils.repeat(adj_factor[:, :, np.newaxis], km + 1, axis=2)
-        final_check(
-            tracers[q],
-            dp2,
-            dm,
-            zfix,
+            sum0,
+            sum1,
             fac,
-            origin=(i1, js, 1),
-            domain=(im, j_extent, km - 1),
+            origin=(i1, js, 0),
+            domain=(im, jext, km),
+            validate_args=False,
```

Comment:
Is there a reason `validate_args=False` here? We should make sure to remove it before merging.

Reply:
Because otherwise the domain computation fails.

Reply:
Removed!